src/Database/IdentifierQuoter.php

```diff
@@ -59,12 +60,14 @@ public function quote(Query $query): Query
         $binder = $query->getValueBinder();
         $query->setValueBinder(null);
 
-        if ($query instanceof InsertQuery) {
+        if ($query instanceof SelectQuery) {
+            $this->_quoteSelect($query);
+        } elseif ($query instanceof InsertQuery) {
             $this->_quoteInsert($query);
         } elseif ($query instanceof UpdateQuery) {
             $this->_quoteUpdate($query);
-        } else {
-            $this->_quoteParts($query);
+        } elseif ($query instanceof DeleteQuery) {
+            $this->_quoteDelete($query);
         }
```

Comment:
should this have an else throw?

Reply:
Guess we could for completeness.

Reply:
Could get fancy and use `match`.

Reply:
Right, can use `match()`, still haven't got accustomed to the fact that it can use expressions for matching :).

Reply:
Done